src/libxively/datastructures/xi_vector.c

```diff
@@ -249,15 +251,18 @@ xi_vector_index_type_t xi_vector_find( xi_vector_t* vector,
     return -1;
 }
 
-void xi_vector_for_each( xi_vector_t* vector, xi_vector_for_t* fun_for )
+void xi_vector_for_each( xi_vector_t* vector,
+                         xi_vector_for_t* fun_for,
+                         void* arg,
+                         xi_vector_index_type_t offset )
 {
     assert( NULL != vector );
 
-    xi_vector_index_type_t i = 0;
+    xi_vector_index_type_t i = offset;
 
```

Comment:
Yes and no, this makes the implementation easier like the border conditions are ```>= 0``` plus returning -1 is a nice option in case of no index found.